hooks/persistence-defectdojo/src/main/java/io/securecodebox/persistence/DefectDojoPersistenceProvider.java

```diff
@@ -17,84 +17,71 @@
+    String findingUploadUrl = args[3];
+
+    var s3Service = new S3Service();
+    var kubernetesService = new KubernetesService();
+    kubernetesService.init();
 
-    var scan = new Scan(getScanFromKubernetes());
+    var scan = new Scan(kubernetesService.getScanFromKubernetes());
     scan.validate();
 
     var config = DefectDojoConfig.fromEnv();
 
-    LOG.info("URL: {}", config.getUrl());
+    LOG.info("Downloading Scan Report (RawResults)");
+    var rawResults = s3Service.downloadRawResults(rawResultDownloadUrl);
+    LOG.info("Finished Downloading Scan Report (RawResults)");
+    LOG.debug("RawResults: {}", rawResults);
+
+    LOG.info("Uploading Findings to DefectDojo at: {}", config.getUrl());
 
     var defectdojoImportStrategy = new VersionedEngagementsStrategy();
     defectdojoImportStrategy.init(config);
+    var defectDojoFindings = defectdojoImportStrategy.run(scan, rawResults);
 
-    defectdojoImportStrategy.run(scan);
-  }
-
-  private static V1Scan getScanFromKubernetes() throws IOException {
-    ApiClient client;
-
-    if ("true".equals(System.getenv("IS_DEV"))) {
-      // loading the out-of-cluster config, a kubeconfig from file-system
-      String kubeConfigPath = System.getProperty("user.home") + "/.kube/config";
-      client = ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(new FileReader(kubeConfigPath))).build();
-    } else {
-      client = ClientBuilder.cluster().build();
-    }
-
-    // set the global default api-client to the in-cluster one from above
-    Configuration.setDefaultApiClient(client);
-
-    String scanName = System.getenv("SCAN_NAME");
-    if (scanName == null) {
-      scanName = "nmap-scanme.nmap.org";
-    }
-    String namespace = System.getenv("NAMESPACE");
-    if (namespace == null) {
-      namespace = "default";
-    }
+    LOG.info("Identified total Number of findings in DefectDojo: {}", defectDojoFindings.size());
 
-    // set the global default api-client to the in-cluster one from above
-    Configuration.setDefaultApiClient(client);
+    var endpointService = new EndpointService(config);
+    var mapper = new DefectDojoFindingToSecureCodeBoxMapper(config, endpointService);
 
-    GenericKubernetesApi<V1Scan, V1ScanList> scanApi =
-      new GenericKubernetesApi<>(
-        V1Scan.class,
-        V1ScanList.class,
-        "execution.securecodebox.io",
-        "v1",
-        "scans",
-        ClientBuilder.defaultClient());
+    var findings = defectDojoFindings.stream()
+      .map(mapper::fromDefectDojoFining)
+      .collect(Collectors.toList());
 
-    var response = scanApi.get(namespace, scanName);
+    LOG.debug("Mapped Findings: {}", findings);
 
-    if (!response.isSuccess()) {
-      throw new DefectDojoPersistenceException("Failed to fetch Scan '" + scanName + "' in Namespace '" + namespace + "' from Kubernetes API");
-    }
-    LOG.info("Fetched Scan from Kubernetes API");
+    s3Service.overwriteFindings(findingUploadUrl, findings);
```

Comment:
Hi @J12934 , i would suggest to make this behavior also configurable by an additional parameter. There might be some situations where it is not needed to write back all defectdojo data to the scb findings 🤷🏻‍♂️. 

```yaml
defectdojo:
  # -- syncBack True to sync back (two way sync) all imported findings from DefectDojo to SCB Findings Store, false (one way sync) otherwise.
  syncBack: true
```

Reply:
Now implemented to handle either case 👍